### PR TITLE
Adopt SPEC 0 policy and drop NEP 29 policy

### PR DESCRIPTION
**Pull request by seisman**
The [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) policy is superseded by the [SPEC 0](https://scientific-python.org/specs/spec-0000/) policy.

Address https://github.com/GenericMappingTools/pygmt/issues/2863.

### CODE REVIEW

File: doc/minversions.rst [hunk 1]
```diff
@@ -1,13 +1,11 @@
 Minimum Supported GMT/Python/NumPy Versions
 -------------------------------------------
```
Comment (weiji14) [inline doc/minversions.rst:2]: Would this be too long?

```suggestion
Minimum Supported GMT/Python/NumPy/Pandas/Xarray Versions
---------------------------------------------------------
```

File: doc/minversions.rst [hunk 2]
```diff
@@ -1,13 +1,11 @@
...
-alongside the rest of the Scientific Python ecosystem, and therefore supports:
+PyGMT has adopted `SPEC 0 <https://scientific-python.org/specs/spec-0000/>`__ alongside
+the rest of the Scientific Python ecosystem, and therefore:
 
-* All minor versions of Python released 42 months prior to the project, and at minimum
-  the two latest minor versions.
-* All minor versions of NumPy released in the 24 months prior to the project, and at
-  minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package dependencies be dropped 2 years after their initial release.
```
Comment (weiji14) [inline doc/minversions.rst:8]: Specify what we mean by core packages
```suggestion
* Support for core package dependencies (NumPy/Pandas/Xarray) be dropped 2 years after their initial release.
```

File: doc/maintenance.md [hunk 3]
```diff
@@ -221,9 +219,8 @@ publishing the actual release notes at {doc}`changes`.
    GitHub accounts. Sort their names by the number of commits made since the
    last release (e.g., use `git shortlog HEAD...v0.4.0 -sne`).
 8. Update `doc/minversions.rst` with new information on the new release version,
-   including a vX.Y.Z documentation link, and minimum required GMT/Python/NumPy
-   versions. Follow
-   [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#detailed-description)
+   including a vX.Y.Z documentation link, and minimum required versions of GMT, Python
+   and core packages. Follow [SPEC 0](https://scientific-python.org/specs/spec-0000/)
```
Comment (weiji14) [inline doc/maintenance.md:223]:
```suggestion
   and core packages (NumPy/Pandas/Xarray). Follow
   [SPEC 0](https://scientific-python.org/specs/spec-0000/)
```

File: doc/maintenance.md [hunk 4]
```diff
@@ -119,17 +119,15 @@ to change the default behaviour at <https://docs.readthedocs.io/en/stable/config
...
-alongside the rest of the Scientific Python ecosystem, and therefore supports:
+PyGMT has adopted [SPEC 0](https://scientific-python.org/specs/spec-0000/) alongside
+the rest of the Scientific Python ecosystem, and therefore:
 
-* All minor versions of Python released 42 months prior to the project,
-  and at minimum the two latest minor versions.
-* All minor versions of NumPy released in the 24 months prior to the project,
-  and at minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package dependencies be dropped 2 years after their initial release.
```
Comment (weiji14) [inline doc/maintenance.md:126]:
```suggestion
* Support for core package (NumPy/Pandas/Xarray) dependencies be dropped 2 years after their initial release.
```

File: README.rst [hunk 5]
```diff
@@ -253,16 +253,15 @@ Other non-official Python wrappers for GMT (not maintained):
...
-alongside the rest of the Scientific Python ecosystem, and therefore supports:
+PyGMT has adopted `SPEC 0 <https://scientific-python.org/specs/spec-0000/>`__ alongside
+the rest of the Scientific Python ecosystem, and therefore:
 
-* All minor versions of Python released 42 months prior to the project, and at minimum
-  the two latest minor versions.
-* All minor versions of NumPy released in the 24 months prior to the project, and at
-  minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package dependencies be dropped 2 years after their initial release.
```
Comment (weiji14) [inline README.rst:263]:
```suggestion
* Support for core package dependencies (NumPy/Pandas/Xarray) be dropped 2 years after their initial release.
```

File: .github/workflows/ci_tests.yaml [hunk 6]
```diff
@@ -17,7 +17,7 @@
 # In draft pull request, only two jobs on Linux are triggered to save on
 # Continuous Integration resources:
 #
-# - Minimum [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy) Python/NumPy versions
+# - Minimum supported Python/NumPy versions following [SPEC 0](https://scientific-python.org/specs/spec-0000/)
```
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:20]:
```suggestion
# - Minimum supported Python/NumPy/Pandas/Xarray versions following [SPEC 0](https://scientific-python.org/specs/spec-0000/)
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 7]
```diff
@@ -16,7 +16,7 @@ assignees: ''
 - [ ] Wrap Y ()
 
 **Before release**:
-- [ ] Check [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) to see if we need to bump the minimum Python and NumPy versions
+- [ ] Check [SPEC 0](https://scientific-python.org/specs/spec-0000/) to see if we need to bump the minimum supported versions of Python and core package dependencies
```
Comment (weiji14) [inline .github/ISSUE_TEMPLATE/release_checklist.md:19]:
```suggestion
- [ ] Check [SPEC 0](https://scientific-python.org/specs/spec-0000/) to see if we need to bump the minimum supported versions of Python and core package dependencies (NumPy/Pandas/Xarray)
```

File: doc/minversions.rst (hunk 1 above)
Comment (seisman) [reply doc/minversions.rst:2]: Perhaps "Minimum Supported Versions" is enough.

File: doc/maintenance.md [hunk 8]
```diff
@@ -119,17 +119,15 @@ to change the default behaviour at <https://docs.readthedocs.io/en/stable/config
...
-alongside the rest of the Scientific Python ecosystem, and therefore supports:
+PyGMT has adopted [SPEC 0](https://scientific-python.org/specs/spec-0000/) alongside
+the rest of the Scientific Python ecosystem, and therefore:
 
-* All minor versions of Python released 42 months prior to the project,
-  and at minimum the two latest minor versions.
-* All minor versions of NumPy released in the 24 months prior to the project,
-  and at minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package (NumPy/Pandas/Xarray) dependencies be dropped 2 years after their initial release.
```
Comment (seisman) [inline doc/maintenance.md:126]:
```suggestion
* Support for core package dependencies (NumPy/Pandas/Xarray) be dropped 2 years after their initial release.
```

File: doc/maintenance.md [hunk 9]
```diff
@@ -221,9 +219,9 @@ publishing the actual release notes at {doc}`changes`.
    GitHub accounts. Sort their names by the number of commits made since the
    last release (e.g., use `git shortlog HEAD...v0.4.0 -sne`).
 8. Update `doc/minversions.rst` with new information on the new release version,
-   including a vX.Y.Z documentation link, and minimum required GMT/Python/NumPy
-   versions. Follow
-   [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#detailed-description)
+   including a vX.Y.Z documentation link, and minimum required versions of GMT, Python
+   and core packages (NumPy/Pandas/Xarray). Follow
```
Comment (seisman) [inline doc/maintenance.md:223]:
```suggestion
   and core package dependencies (NumPy/Pandas/Xarray). Follow
```

**Review by michaelgrund — APPROVED** [review]

File: doc/minversions.rst (hunk 1 above)
Comment (seisman) [reply doc/minversions.rst]: Changed to "Minimum Supported Versions" in f4049a09c9206c5b80cb47d56521453f1a16cbb4.

File: README.rst [hunk 10]
```diff
@@ -253,16 +253,15 @@ Other non-official Python wrappers for GMT (not maintained):
 
 .. doc-index-end-before
 
-Minimum Supported GMT/Python/NumPy Versions
--------------------------------------------
+Minimum Supported Versions of GMT/Python/Core package dependencies
+------------------------------------------------------------------
```
Comment (yvonnefroehlich) [inline README.rst:257]: For consistency with the external file `minversions.rst`:

```suggestion
Minimum Supported Versions
--------------------------
```

File: README.rst [hunk 11]
```diff
@@ -253,16 +253,15 @@ Other non-official Python wrappers for GMT (not maintained):
...
-* All minor versions of NumPy released in the 24 months prior to the project, and at
-  minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package dependencies (NumPy/Pandas/Xarray) be dropped 2 years after their initial release.
 
-For the supported GMT versions as well as the minimum required Python and NumPy versions
-please see `Minimum Required Versions <https://www.pygmt.org/dev/minversions.html>`__.
+For the supported GMT versions as well as the minimum required versions of Python and
+core package dependencies, please see
+`Minimum Required Versions <https://www.pygmt.org/dev/minversions.html>`__.
```
Comment (yvonnefroehlich) [inline README.rst:267]: To be consistent with the title used in the `README.rst` and `minversions.rst`.

```suggestion
`Minimum Supported Versions <https://www.pygmt.org/dev/minversions.html>`__.
```

File: doc/maintenance.md [hunk 12]
```diff
@@ -221,9 +219,9 @@ publishing the actual release notes at {doc}`changes`.
    GitHub accounts. Sort their names by the number of commits made since the
    last release (e.g., use `git shortlog HEAD...v0.4.0 -sne`).
 8. Update `doc/minversions.rst` with new information on the new release version,
-   including a vX.Y.Z documentation link, and minimum required GMT/Python/NumPy
-   versions. Follow
-   [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#detailed-description)
+   including a vX.Y.Z documentation link, and minimum required versions of GMT, Python
+   and core package dependencies (NumPy/Pandas/Xarray). Follow
+   [SPEC 0](https://scientific-python.org/specs/spec-0000/)
    for updates.
```
Comment (yvonnefroehlich) [inline doc/maintenance.md:225]: I think this should fit into one line.
```suggestion
   [SPEC 0](https://scientific-python.org/specs/spec-0000/) for updates.
```

File: doc/minversions.rst [hunk 13]
```diff
@@ -1,13 +1,11 @@
...
 
-* All minor versions of Python released 42 months prior to the project, and at minimum
-  the two latest minor versions.
-* All minor versions of NumPy released in the 24 months prior to the project, and at
-  minimum the last three minor versions.
+* Support for Python versions be dropped 3 years after their initial release.
+* Support for core package dependencies (NumPy/Pandas/Xarray) be dropped 2 years after their initial release.
 
 .. list-table::
     :widths: 25 30 15 20 15
```
Comment (weiji14) [inline doc/minversions.rst:11]: Do we want to add columns for Pandas and Xarray in this PR, or in a follow up PR?

**Review by weiji14 — APPROVED** [review]

**Pull request merged**